pkg/server/delta/v2/server.go

```diff
@@ -0,0 +1,49 @@
+package delta
+
+import (
+	"context"
+
+	discovery "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v2"
+	"github.com/envoyproxy/go-control-plane/pkg/server/stream/v2"
+)
+
+// Server is defined to implement the specific stream handler type
+type Server interface {
+	DeltaStreamHandler(stream stream.DeltaStream, typeURL string) error
+}
+
+type Callbacks interface {
+	// OnDeltaStreamOpen is called once an incremental xDS stream is open with a stream ID and the type URL (or "" for ADS).
+	// Returning an error will end processing and close the stream. OnStreamClosed will still be called.
+	OnDeltaStreamOpen(context.Context, int64, string) error
+	// OnDeltaStreamClosed is called immediately prior to closing an xDS stream with a stream ID.
+	OnDeltaStreamClosed(int64)
+	// OnStreamDeltaRequest is called once a request is received on a stream.
+	// Returning an error will end processing and close the stream. OnStreamClosed will still be called.
+	OnStreamDeltaRequest(int64, *discovery.DeltaDiscoveryRequest) error
+	// OnStreamDelatResponse is called immediately prior to sending a response on a stream.
+	OnStreamDeltaResponse(int64, *discovery.DeltaDiscoveryRequest, *discovery.DeltaDiscoveryResponse)
+}
+
+type server struct {
+	cache     cache.ConfigWatcher
+	callbacks Callbacks
+
+	// streamCount for counting bi-di streams
+	streamCount int64
+	ctx         context.Context
+}
+
+// NewServer creates handlers from a config watcher and callbacks.
```

Comment:
it creates a server right? not handlers

Reply:
Ah sorry, I copied that comment from the SOTW implementation. Will clarify